### PR TITLE
ZDC: add post-processing for ADC and TDC plots

### PR DESCRIPTION
**Pull request by aferrero2707**
The post-processing is needed in order to correctly plot the average values and errors from individual ADC and TDC plots.
The reference JSON configuration is also updated accordingly, with some improvements:
- added post-processing tasks
- updated checker to take post-processing output and use `OnEachSeparately` policy
- added quality aggregator and post-processing (summary)

### CODE REVIEW

File: Modules/ZDC/include/ZDC/ZDCRecDataPostProcessing.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+/// \brief   Post-processing of the MCH pre-clusters
+/// \since   21/06/2022
+///
+
+#ifndef QC_MODULE_ZDC_ZDCZDCRECDATAPP_H
+#define QC_MODULE_ZDC_ZDCZDCRECDATAPP_H
+
+#include "QualityControl/PostProcessingInterface.h"
+
+#include <TH1F.h>
```
Comment (knopers8) [inline Modules/ZDC/include/ZDC/ZDCRecDataPostProcessing.h:24]:
```suggestion
#include <TH1F.h>
#include <memory>
#include <string>
```

File: Modules/ZDC/src/ZDCRecDataPostProcessing.cxx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    ZDCRecDataPostProcessing.cxx
+/// \author  Andrea Ferrero andrea.ferrero@cern.ch
+/// \brief   Post-processing of the MCH pre-clusters
```
Comment (knopers8) [inline Modules/ZDC/src/ZDCRecDataPostProcessing.cxx:15]:
```suggestion
/// \brief   Post-processing of the ZDC ADC and TDC plots
```

File: Modules/ZDC/src/ZDCRecDataPostProcessing.cxx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+  return timeStamp;
+}
+
+//_________________________________________________________________________________________
+
+static bool checkMoTimeStamp(std::shared_ptr<o2::quality_control::core::MonitorObject> mo, uint64_t& timeStampOld)
+{
+  auto timeStamp = getMoTimeStamp(mo);
+  bool result = (timeStamp != timeStampOld) ? true : false;
```
Comment (knopers8) [inline Modules/ZDC/src/ZDCRecDataPostProcessing.cxx:60]:
```suggestion
  bool result = timeStamp != timeStampOld;
```

**Review by knopers8 — APPROVED** [review]
Thank you for offering your help to address this ZDC issue.

I have some editorial suggestions, but I think it's good as it is.

File: Modules/ZDC/src/ZDCRecDataPostProcessing.cxx (hunk 3 above)
Comment (aferrero2707) [reply Modules/ZDC/src/ZDCRecDataPostProcessing.cxx]: Done.

File: Modules/ZDC/include/ZDC/ZDCRecDataPostProcessing.h (hunk 1 above)
Comment (aferrero2707) [reply Modules/ZDC/include/ZDC/ZDCRecDataPostProcessing.h:24]: Done.

File: Modules/ZDC/src/ZDCRecDataPostProcessing.cxx (hunk 2 above)
Comment (aferrero2707) [reply Modules/ZDC/src/ZDCRecDataPostProcessing.cxx]: Corrected, including few more leftovers from the MCH code used as starting point.

**Comment by aferrero2707** [comment]
@knopers8 I spotted a typo in the original names of the summary plots (triple "m" in the name), I fixed it in order to start the HI period with consistent naming.
Unfortunately this means you will have to re-enable the automatic merging...

Thanks!

**Pull request merged**